homework/shared_ptr/shared_ptr.hpp

```diff
@@ -0,0 +1,164 @@
+#pragma once
+#include <atomic>
+
```

Comment:
Please put your implementation in `my` namespace to avoid confusion with `std::shared_ptr`. This is required by our CI job.

```suggestion
namespace my {

```